### PR TITLE
fix: Make sonar exclusions a variable to reuse this file

### PR DESCRIPTION
**Pull request by AsabuHere**
# Fixes #
DII-640
A short description of what this PR does.
Make sonar exclusions a variable to reuse this file and run with different coverage areas
### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-csharp/blob/main/CONTRIBUTING.md) and my PR follows them
- [X] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

**Review by charan678 — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -35,7 +35,7 @@ docker-push:
 	docker push twilio/twilio-csharp:${CURRENT_TAG}
 
 cover:
-	dotnet sonarscanner begin /k:"$(PROJECT_NAME)" /o:"twilio" /d:sonar.host.url=https://sonarcloud.io /d:sonar.login="${SONAR_TOKEN}"  /d:sonar.language="cs" /d:sonar.exclusions="src/Twilio/Rest/**/*.*,test/Twilio.Test/**/*.*" /d:sonar.cs.opencover.reportsPaths="test/lcov.net451.opencover.xml"
+	dotnet sonarscanner begin /k:"$(PROJECT_NAME)" /o:"twilio" /d:sonar.host.url=https://sonarcloud.io /d:sonar.login="${SONAR_TOKEN}"  /d:sonar.language="cs" $(SONAR_SOURCES) /d:sonar.cs.opencover.reportsPaths="test/lcov.net451.opencover.xml"
```
Comment (childish-sambino) [inline Makefile:38]: Prefer to have a default value defined in this file for `SONAR_SOURCES` which can be overridden. E.g., https://stackoverflow.com/a/24264930

Comment (childish-sambino) [reply Makefile:38]: I think the same should go for the `PROJECT_NAME`.

Comment (AsabuHere) [reply Makefile:40]: Added default values for both PROJECT_NAME and SONAR_SOURCES

File: .github/workflows/test-and-deploy.yml [hunk 2]
```diff
@@ -36,7 +36,7 @@ jobs:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
         run: |
           dotnet tool install --global dotnet-sonarscanner
-          make cover PROJECT_NAME=twilio_twilio-csharp
+          make cover PROJECT_NAME=twilio_twilio-csharp SONAR_SOURCES="/d:sonar.exclusions=src/Twilio/Rest/**/*.*,test/Twilio.Test/**/*.*"
```
Comment (childish-sambino) [inline .github/workflows/test-and-deploy.yml:39]: These env vars can be removed from this line since they're using the defaults.

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**